python/prepare_release.py

```diff
@@ -0,0 +1,296 @@
+                                                                            upcoming_minor_version, upcoming_version)
+        # find current schema version info and update it with upcoming version  in multi_extension.out file
+        update_schema_version_with_upcoming_version_in_multi_extension_file(current_schema_version,
+                                                                            multi_extension_out_path,
+                                                                            upcoming_minor_version, upcoming_version)
+        # create new sql file with the name ">./src/backend/distributed/citus--$current_schema_version--
+        # $upcoming_minor_version-1.sql"
+        newly_created_sql_file = create_new_upcoming_version_sql_file(current_schema_version, distributed_dir_path,
+                                                                      upcoming_minor_version)
+        # change version in citus.control file
+        update_version_with_upcoming_version_in_citus_control(citus_control_file_path, upcoming_minor_version)
+        # commit and push changes on master-update-version-$curtime branch
+        commit_all_changes_on_branch(project_name, project_version, upcoming_version_branch)
+        if not is_test:
+            push_branch(upcoming_version_branch)
+            resource_status = ResourceStatus.UPCOMING_VERSION_REMOTE
+        # create pull request
+        if not is_test:
+            create_pull_request_for_upcoming_version_branch(main_branch, repository, upcoming_version,
+                                                            upcoming_version_branch)
+            resource_status = ResourceStatus.PULL_REQUEST_CREATED
+    else:
+        print(f"### {project_version} is a patch release. Executing Patch release flow ### ")
+        # checkout release branch in release-X.Y format
+        checkout_branch(release_branch_name, is_test)
+
+        # change version info in configure.in file
+        update_version_in_configure_in(configure_in_path, project_version)
+        # execute "auto-conf " (Not with f flag!!!)
+        execute_autoconf_f()
+        # change version info in multi_extension.out
+        update_version_in_multi_extension_out(multi_extension_out_path, project_version)
+
+        if cherry_pick_enabled:
+            # list all pr's with backport labels
+            list_and_cherrypick_prs_with_backport_labels(earliest_pr_date, main_branch, release_branch_name, repository)
+
+        # commit all-changes
+        commit_all_changes_on_branch(project_name, project_version, release_branch_name)
+        # create and push release-$minor_version-push-$curTime branch
+        release_pr_branch = f"{release_branch_name}_{uuid.uuid4()}"
+        create_branch(release_pr_branch)
+        if not is_test:
+            push_branch(release_pr_branch)
+    return release_branch_name, upcoming_version_branch, f"{DISTRIBUTED_DIR_PATH}/{newly_created_sql_file}", \
+           resource_status
```

Comment:
Instead of returning a tuple, can we define a class such as `UpdateReleaseReturnValue` ?

Reply:
Fixed